lib/libzfs/libzfs_mount.c

```diff
@@ -972,9 +972,21 @@ zfs_unshareall(zfs_handle_t *zhp)
 }
 
 int
-zfs_unshareall_bypath(zfs_handle_t *zhp, const char *mountpoint)
+zfs_unshareall_bypath(zfs_handle_t *zhp, const char *mountpoint,
+    const char *proto)
 {
-	return (zfs_unshare_proto(zhp, mountpoint, share_all_proto));
+	zfs_share_proto_t *curr_proto = NULL;
```

Comment:
Maybe a better name for `curr_proto`, or even better just remove the need for the variable entirely and call `return zfs_unshare_proto()` in the conditional.